src/ucp/proto/proto_am.inl

```diff
@@ -174,6 +174,8 @@ ucs_status_t ucp_do_am_zcopy_single(uct_pending_req_t *self, uint8_t am_id,
         req->send.state = saved_state; /* need to restore the offsets state */
         return status;
     } else {
+        ++req->send.uct_comp.count;
```

Comment:
i would rework it as well with common functions :)
it can be implemented w/o overhead using inline + constant arg for different protocols